docs/multicall-notes.md

```diff
@@ -0,0 +1,225 @@
+
+### Phantom blocks
+The multicall allows you to define on what block number your calls or transactions are being executed on. E.g, consider following call:
+```json
+{
+	"jsonrpc": "2.0",
+	"id": 1,
+	"method": "eth_multicallV1",
+	"params": [
+		{
+			"blockStateCalls": [
+				{
+					"blockOverrides": {
+						"number": "0x64"
+					},
+				},
+				{
+					"blockOverrides": {
+						"number": "0xc8"
+					},
+				}
+			]
+		},
+		"0xa"
+	]
+}
+```
+
+Here we want our calls to be executed in blocks 100 (`0x64`) and in 200 (`0xc8`). The block numbers can be anything as long as they are increasing and higher than the block we are building from 10 (`0xa`). Now we end up in a situation where there exists block ranges 13-99 and 101-199 that are not defined anywhere. These blocks are called "phantom blocks". What happens if you try to request block hash of any of such blocks in the EVM? How can we calculate the block hash of future blocks when we don't know the block hash of the previous block?
+
+Our solution to this problem is to define block hash of a phantom block to be:
+
+```
+keccak(rlp([hash_of_previous_non_phantom_block, phantom_block_number]))
+```
+
+So for example in our example, you could get block hash of block 142 as follows: 
+```
+keccac(rlp([hash of block 12, 142]))
+```
+
+The phantom blocks other properties are set to their default properties as defined by the multicall specification. We came to this definition by wanting phantom block hashes to be unique if things prior to the phantom block changes, so if tooling is storing block hashes somewhere, they should remain unique if things change in the simulation.
+
+One other approach to this problem would be to really calculate the real block hashes for all the phantom blocks, but this would make generating blocks far in future really expensive, as to generate 100 phantom blocks, you would need to calculate 100 block hashes that all depend on each other. And in most cases, no one really cares about these blocks.
+
+Base fee per gas is not adjusted in the phantom blocks, their base fee remains constant.
+
+## Default values for transactions
+As multicall is an extension to `eth_call` we want to enable the nice user experience that the user does not need to provide all required values for a transaction. We are assuming following defaults if the variable is not provided by the user:
+| parameter name | description |
+-----------------|-----------------------
+| type | `0x2` |
+| nonce | Defaults to correct nonce |
+| to | `null` |
+| from | `0x0000000000000000000000000000000000000000` |
+| gas limit | Remaining gas in the current block. This is calculated dynamically one by one for each transaction that is being processed. |
+| value | `0x0` |
+| input | no data |
+| gasPrice | `0x0` |
+| maxPriorityFeePerGas | `0x0` |
+| maxFeePerGas | `0x0` |
```

Comment:
These default values will crash, since baseFeePerGas is always 7 or higher. These default values will be rejected by blockchain spec, since tx is not willing to pay base fee. Note that the default block `baseFeePerGas` is calculated according to spec, so this will be 7 or higher always.

Reply:
Suggestion: change `maxFeePerGas` and `gasPrice` (in case of tx type 0/1) to `baseFeePerGas` of the block and keep `maxPriorityFeePerGas` 0.

Reply:
> These default values will crash, since baseFeePerGas is always 7 or higher. These default values will be rejected by blockchain spec, since tx is not willing to pay base fee. Note that the default block `baseFeePerGas` is calculated according to spec, so this will be 7 or higher always.

yeah, this will crash on validation mode. On non-validation mode, we will allow you to make transactions that have zero basefee (similar to eth_call)